apis/v1alpha1/gateway_types.go

```diff
@@ -205,9 +198,8 @@ type HostnameMatch struct {
 	// Match specifies how the hostname provided by the client should be
 	// matched against the given value.
 	//
-	// +optional
 	// +kubebuilder:default=Exact
```

Comment:
@robscott I question whether "Exact" is the correct default due to the `Name` field details:
```
<SNIP>
// This field is required for the "Domain" and "Exact" match types.
//
// +optional
// +kubebuilder:validation:MaxLength=253
Name string `json:"name"`
```

`mach: Exact` will be defaulted which requires `Name` but name is optional for this match type. Maybe the default should be "Any". Thoughts? Let me know if this issue should be addressed separately.

Reply:
@jpeach Do you recall why we settled on Exact in the past? It probably warrants its own issue.

Reply:
https://github.com/kubernetes-sigs/service-apis/issues/348